### DISCUSSION TITLE
How to deploy binaries (iconv)?

### DISCUSSION THREAD

**Discussion by riabiy** [hammerstonedev/sidecar#114] [Q&A] [answered]
One of the dependencies of the package I need to use in Sidecar is `iconv`. 
This is a binary, from what I can tell. So, if I build it locally (`npm ci`), I can't deploy it to lambda. Well, I can, but it won't work.

What is the recommended strategy to deal with that? 

Actual error
```
node_modules/iconv/build/Release/iconv.node: invalid ELF header.
```

**Comment by aarondfrancis** [comment]
I think you have to build that particular binary for Lambda by using a Lambda-like environment. 

Here's an article on how to do it for Pandoc, but it should be similar: https://claudiajs.com/tutorials/pandoc-lambda.html.

**Comment by riabiy** [comment, marked as answer]
@aarondfrancis thank you. Overall, I agree with you. 

## Followup for everyone else
For anyone searching around -> we've just did a few hacks to have a JS-verson, that avoids compiling. This is the first time facing compiling in JS and, I hope, the last one :D 

Overall, we were able to compile for lambda. Technically, it's possible. So, if you really need it and wanna know, if this is ever gonna work -> yes. In our case, when we had an Intel chip -> it was relatively easy. But after getting Apple Silicone -> things got harder. 

Soo...my 2 cents. Just avoid it. Better to spend 4 hours vendor sourcing, than 4 hours compiling and then 4 hours vendor-sourcing 😆
